src/replica/replica_stub.cpp

```diff
@@ -618,6 +618,10 @@ void replica_stub::initialize(const replication_options &opts, bool clear /* = f
     LOG_INFO("start to load replicas");
     std::map<dir_node *, std::vector<std::string>> dirs_by_dn;
     for (const auto &dn : _fs_manager.get_dir_nodes()) {
+        // Skip IO error dir_node.
+        if (dn->status == disk_status::IO_ERROR) {
```

Comment:
The reason why do not check `dn->status != disk_status::NORMAL` is that even if the status is `disk_status::SPACE_INSUFFICIENT` we could still just read this disk rather than writing into it ? Once current status is `disk_status::SPACE_INSUFFICIENT` should we reject any write ?

Reply:
Yes.
It is how we do in replica::on_client_write.